audbackend/core/backend.py

```diff
@@ -132,11 +134,12 @@ def get_archive(
 
         Examples:
             >>> dst_root = audeer.path(tmp, 'dst')
-            >>> backend.get_archive('folder/name', dst_root, '1.0.0')
+            >>> backend.get_archive('folder/name.zip', dst_root, '1.0.0')
             ['src.pth']
 
         """
-        src_path += '.zip'
+        if not src_path.endswith('.zip'):
+            src_path += '.zip'
```

Comment:
Would it make more sense to raise an error if the extension does not match?

I ask, because it can happen that you don't find an error if you make a small typo, e.g.

```python
>>> zip_file = 'archive.Zip'
>>> backend.put_archive('.', ['name.txt'], zip_file, '1.0.0')
>>> backend.exists(zip_file, '1.0.0')
False
```

Reply:
In that case we have to:

![image](https://user-images.githubusercontent.com/10383417/231692119-958f4208-bea9-4793-b530-54332daea8f9.png)

as mentioned in https://github.com/audeering/audbackend/issues/78#issuecomment-1506479093

Reply:
Yes, but we have to update all of them anyway as the `ext` argument is removed and we have breaking changes.

Reply:
In `audb` we do not use `ext` and also not `glob()` . But I just checked and in `audb.available()` we use `ls()`, which now has a new return type. So we indeed will break all packages anyway. I.e. it would indeed not matter if we add another breaking change on top.

Reply:
I'm not completely sure what is the best option:

* Automatically add `.zip` if file extension does not match. Disadvantage: `.Zip` typos might be unnoticed
* Raise an error if file extension is not `.zip`. Disadvantage: user might wonder why she has to provide a file extension, but cannot choose a different one
* Don't provide file extension and always append `.zip`. Disadvantage: feels unintuitive as we provide file extension to all other methods, e.g. `backend.exists('archive.zip', '1.0.0')`
* Allow for arbitrary or no file extension. Disadvantage: ZIP files can be stored on the backend with "wrong" file extensions and might not be supported by backend functionality like web UIs.

So it seems everything has a disadvantage ;)

Reply:
In `audeer` we currently support ZIP and TAR.GZ and raise an error if the archive name does not end on either `zip` or `tar.gz`:

![image](https://user-images.githubusercontent.com/10383417/231695259-2ca335b2-4c93-44c2-af61-22b5b20e555b.png)

So maybe we should indeed do the same here.

Reply:
Ok, let's simply not check for the file extension and let `audeer` do the work for us. In the docstring we then state that we support the archives as `audeer.create_archive()` and that an error is raised if a non-supported file extension is used. This also has the advantage that we will automatically support all archives types that can be handeled by `audeer`:

Reply:
OK, at least we then have another example with `audeer` where we do the same. So yes, select 2. and raise an error.

I guess more general programs/modules simply allow for arbitrary file extensions when creating the ZIP file, but as we specifically target backend it might be a good idea to force the correct name as you also indicated.

Reply:
O, let's go with your suggestion and let `audeer` handle it.

Reply:
done

Reply:
Cool, thanks.